build/config/compiler.gni

```diff
@@ -5,6 +5,14 @@
 
 # Adjusts compiler flags in //build/config/compiler:chromium_code.
 brave_removed_chromium_code_cflags = []
+brave_removed_compiler_cflags = []
+brave_added_compiler_cflags = []
 if (is_clang) {
   brave_removed_chromium_code_cflags += [ "-Wunreachable-code" ]
+  if (is_android) {
+    # We need to remove combination of "-mllvm" and "-enable-dse-memoryssa=false",
```

Comment:
this change concerns me, why are we having the issue if upstream is not?

Reply:
Yes, this requires research, I just found commit that broke the build
I thought it would be safe to remove this flag, since they reverted clang revision for which that flag was added https://github.com/chromium/chromium/commit/1bc47ee270b21837f4321a9c8fea0e8f5690a2c5
Alternatively, we may set `use_thin_lto = false`, till we figure out why error happens or till they remove that flag in upstream (as it is marked to be removed)

Reply:
There is no linker error in `cr88` btw, I believe setting `use_thin_lto = false` is the safest way to workaround it for now and I will create a follow up issue to research that error. @bridiver does it sound ok?

Reply:
This error is actually related to `sccache`, after cleaning it up there is no error anymore. I will revert this change.

Reply:
Revrted https://github.com/brave/brave-core/commit/42498157ef2780c4dfb4495f4412ee12ea94769c